File: Amethyst/NSRunningApplication+Manageable.m
//
//  NSRunningApplication+Manageable.m
//  Amethyst
//
//  Created by Ian Ynda-Hummel on 5/24/13.
//  Copyright (c) 2013 Ian Ynda-Hummel. All rights reserved.
//

#import "NSRunningApplication+Manageable.h"

@implementation NSRunningApplication (Manageable)

- (BOOL)isManageable {
    if ([self.bundleIdentifier hasPrefix:@"com.apple.dashboard"]) return NO;
    if ([self.bundleIdentifier hasPrefix:@"com.apple.loginwindow"]) return NO;
    if (self.isAgent) return NO;

    return YES;
}

@end

File: Amethyst/NSRunningApplication+Manageable.swift
//
//  NSRunningApplication+Manageable.swift
//  Amethyst
//
//  Created by Ian Ynda-Hummel on 5/8/16.
//  Copyright © 2016 Ian Ynda-Hummel. All rights reserved.
//

import Foundation

public extension NSRunningApplication {
    public var isManageable: Bool {
        guard let bundleIdentifier = bundleIdentifier where !isAgent() else {
            return false
        }

        switch bundleIdentifier {
        case "com.apple.dashboard":
            return false
        case "com.apple.loginwindow":
            return false
        default:
            return true
        }
    }
}
